hw/ip/hmac/data/hmac.hjson

```diff
@@ -227,80 +227,89 @@
               desc: '''
-              4'b1000: SHA-2 512 digest.
+              4'b1000: Unsupported/invalid values and all-zero values are mapped to SHA2_None.
+              In this case, HMAC/SHA-2 does not start the operation and an error is flagged to SW.
               '''
             },
           ]
           tags: [// Exclude fields with non-zero reset value from write-read checks.
                "excl:CsrNonInitTests:CsrExclWriteCheck"]
         }
-        { bits: "12:8",
+        { bits: "13:8",
           name: "key_length",
-          resval: "0x02",
+          resval: "0x20",
           desc: '''Key length configuration.
 
-                This is a 5-bit one-hot encoded field to configure the key length for HMAC.
-                The HMAC supports key lengths of 128-bit, 256-bit, 384-bit, 512-bit and 1024-bit, but will to block size if key length is greater than block size, i.e. max of 1024 bits for digest size SHA-2 384/512 and respectively 512 bits for SHA-2 256.
+                This is a 6-bit one-hot encoded field to configure the key length for HMAC.
+                The HMAC supports key lengths of 128-bit, 256-bit, 384-bit, 512-bit and 1024-bit, but so long as key length is not greater than the block size of the SHA-2 digest size, i.e. max of 1024 bits for digest size SHA-2 384/512 and respectively 512 bits for SHA-2 256.
+                Otherwise, an error is flagged to SW.
                 The value of this register is irrelevant when only SHA-2 (not keyed HMAC) is configured.
                 '''
           enum: [
             { value: "1",
               name: "Key_128",
               desc: '''
-              5'b0_0001: 128-bit secret key.
+              6'b00_0001: 128-bit secret key.
               '''
             },
             { value: "2",
               name: "Key_256",
               desc: '''
-              5'b0_0010: 256-bit secret key. Unsupported/invalid values and all-zero values
-              are mapped to Key_256.
+              6'b00_0010: 256-bit secret key.
               '''
             },
             { value: "4",
               name: "Key_384",
               desc: '''
-              5'b0_0100: 384-bit secret key.
+              6'b00_0100: 384-bit secret key.
               '''
             },
             { value: "8",
               name: "Key_512",
               desc: '''
-              5'b0_1000: 512-bit secret key.
+              6'b00_1000: 512-bit secret key.
               '''
             },
             { value: "16",
               name: "Key_1024",
               desc: '''
-              5'b1_0000: 1024-bit secret key.
+              6'b01_0000: 1024-bit secret key.
+              '''
+            },
+            { value: "32",
+              name: "Key_None",
+              desc: '''
+              6'b10_0000: Unsupported/invalid values and all-zero values are mapped to Key_None.
+              In this case, HMAC does not start the operation and an error is flagged to SW.
```

Comment:
An error will be flagged if we are in Key_None mode and hamc_en=1?

Reply:
The description above says "The value of this register is irrelevant when only SHA-2 (not keyed HMAC) is configured.", so I would expect that nothing happens if this register is set to an invalid value while `hmac_en == 0`. But I agree that this should be clarified.

Reply:
That's correct, if hmac_en == 0, an error is **not** flagged if the key_length is Key_None, but will clarify further in the description.